src/projections/tobmerc.cpp

```diff
@@ -9,27 +9,24 @@
 
 PROJ_HEAD(tobmerc, "Tobler-Mercator") "\n\tCyl, Sph";
 
-#define EPS10 1.e-10
-static double logtanpfpim1(double x) {       /* log(tan(x/2 + M_FORTPI)) */
-    if (fabs(x) <= DBL_EPSILON) {
-        /* tan(M_FORTPI + .5 * x) can be approximated by  1.0 + x */
-        return log1p(x);
-    }
-    return log(tan(M_FORTPI + .5 * x));
-}
-
 static PJ_XY tobmerc_s_forward (PJ_LP lp, PJ *P) {           /* Spheroidal, forward */
     PJ_XY xy = {0.0, 0.0};
     double cosphi;
 
-    if (fabs(fabs(lp.phi) - M_HALFPI) <= EPS10) {
+    if (fabs(lp.phi) >= M_HALFPI) {
+      // builtins.gie tests "Test expected failure at the poles:".  However
+      // given that M_HALFPI is strictly less than pi/2 in double precision,
+      // it's not clear why shouldn't just return a large result for xy.y (and
+      // it's not even that large, merely 38.025...).  Even if the logic was
+      // such that phi was strictly equal to pi/2, allowing xy.y = inf would be
```

Comment:
I'm not sure returning inf as a valid value is a good idea (especially since it is our HUGE_VAL error marker).

Reply:
I think that makes #2376 more relevant. Would help differentiate between HUGE_VAL that are valid versus invalid.

Reply:
I agree with @snowman2.  However, the issue is moot.  An inf return isn't on the cards in the present code base, because there's no way that phi can be set to exactly 90deg.  The closest is round(pi/2) which then gives asinh(tan(phi)) = 38, a distinctly non-large number.  Things would change, if the representation of angles were changed (e.g., after we get sinpi and cospi functions) so that the cardinal points were exactly represented.  I don't see this happening in the short term.